core/src/main/java/com/novoda/noplayer/NoPlayer.java

```diff
@@ -311,12 +312,16 @@
 
     interface AdvertListener {
 
+        void onAdvertsLoadError(Exception cause);
+
         void onAdvertsLoaded(List<AdvertBreak> advertBreaks);
 
         void onAdvertBreakStart(AdvertBreak advertBreak);
 
         void onAdvertBreakEnd(AdvertBreak advertBreak);
 
+        void onAdvertPrepareError(Advert advert, IOException failedWith);
```

Comment:
Why not just use `cause` instead of `failedWith`?

Also... how about adding some javadoc? 😝